File: packages/core/src/styles/brutalist.ts
import { fnv1a } from '../hash';
import { extractInitials } from '../initials';
import { BRUTALIST_PALETTES } from '../palettes';
import type { BrutalistSpec } from '../types';

export function brutalistStyle(name: string): BrutalistSpec {
	const initials = extractInitials(name);
	const h = fnv1a(name.trim().toLowerCase());
	const palette = BRUTALIST_PALETTES[h % BRUTALIST_PALETTES.length];

	return {
		style: 'brutalist',
		initials,
		palette,
		fontSize: initials.length === 1 ? 80 : 64,
	};
}

File: packages/dart/lib/src/styles/brutalist.dart
import '../hash.dart';
import '../initials.dart';
import '../palettes.dart';
import '../types.dart';

BrutalistSpec brutalistStyle(String name) {
  final initials = extractInitials(name);
  final h = fnv1a(name.trim().toLowerCase());
  final p = brutalistPalettes[h % brutalistPalettes.length];

  return BrutalistSpec(
    initials: initials,
    bg: p['bg']!,
    fg: p['fg']!,
    fontSize: initials.length == 1 ? 80 : 64,
  );
}
